### PR TITLE
Improved docs and optimised

### PR DESCRIPTION
**Pull request by UP929312**
### Describe your change:

* [ ] Add an algorithm?
* [x] Fix a bug or typo in an existing algorithm?
* [x] Documentation change?

### Checklist:
* [x] I have read [CONTRIBUTING.md](https://github.com/TheAlgorithms/Python/blob/master/CONTRIBUTING.md).
* [x] This pull request is all my own work -- I have not plagiarized.
* [x] I know that pull requests will not be merged if they fail the automated tests.
* [x] This PR only changes one algorithm file.  To ease review, please open separate PRs for separate algorithms.
* [x] All new Python files are placed inside an existing directory.
* [x] All filenames are in all lowercase characters with no spaces or dashes.
* [x] All functions and variable names follow Python naming conventions.
* [x] All function parameters and return values are annotated with Python [type hints](https://docs.python.org/3/library/typing.html).
* [x] All functions have [doctests](https://docs.python.org/3/library/doctest.html) that pass the automated testing.
* [x] All new algorithms have a URL in its comments that points to Wikipedia or other similar explanation.
* [x] If this pull request resolves one or more open issues then the commit message contains `Fixes: #{$ISSUE_NO}`.

### CODE REVIEW

File: data_structures/stacks/stack.py [hunk 1]
```diff
@@ -66,17 +66,18 @@ def peek(self) -> T:
 
     def is_empty(self) -> bool:
         """Check if a stack is empty."""
-        return not bool(self.stack)
+        return (not self.stack)
```
Comment (cclauss) [inline data_structures/stacks/stack.py:69]:
```suggestion
        return not self.stack
```

File: data_structures/stacks/stack.py [hunk 2]
```diff
@@ -66,17 +66,18 @@ def peek(self) -> T:
...
 
     def is_full(self) -> bool:
+        """Check if a stack is full."""
         return self.size() == self.limit
 
     def size(self) -> int:
```
Comment (cclauss) [inline data_structures/stacks/stack.py:75]: Please change to
```suggestion
    def __len__(self) -> int:
```
https://docs.python.org/3/reference/datamodel.html#object.__len__ and then change all calls to `.size()`

File: data_structures/stacks/stack.py [hunk 3]
```diff
@@ -66,17 +66,18 @@ def peek(self) -> T:
...
         """Return the size of the stack."""
         return len(self.stack)
 
     def __contains__(self, item: T) -> bool:
```
Comment (cclauss) [inline data_structures/stacks/stack.py:79]: Please sort all the dunder `__functions__()` before the normal functions.

**Review by cclauss — CHANGES_REQUESTED** [review]

File: data_structures/stacks/stack.py (hunk 2 above)
Comment (UP929312) [reply data_structures/stacks/stack.py:75]: Message to the bot: I didn't want to change the actual code that much, but this isn't really a bad idea, just thought I'd keep it the way it was

File: data_structures/stacks/stack.py (hunk 3 above)
Comment (UP929312) [reply data_structures/stacks/stack.py:79]: Meh

File: data_structures/stacks/stack.py (hunk 2 above)
Comment (cclauss) [reply data_structures/stacks/stack.py:75]: Not a bot.

**Pull request closed without merging**